qiskit/providers/aer/quantum_info/states/aer_state.py

```diff
@@ -0,0 +1,297 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2020, 2021, 2022
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+State class that handles internal C++ state safely
+"""
+import numpy as np
+from qiskit.providers.aer.backends.controller_wrappers import AerStateWrapper
+from ...backends.aerbackend import AerError
+
+
+class AerState:
```

Comment:
This really could use some docs to explain how to use this with some examples.

Reply:
I do not expect that end-users use `AerState` directly. In such case, do we need docs?

Reply:
Ok, if it's an internal class we don't need a full docstring then since it won't be published on the docs page. But having some minimal docstrings will help us as most IDEs will be able to read docstrings and present them to developers. So having some minimal explanation of the purpose of the class and some explanation for the methods would be good. You can also mention it's for internal aer use only in the docstring to make it clear